transport/src/main/java/io/servicefabric/transport/ITransport.java

```diff
@@ -19,14 +23,64 @@
   void start();
 
   /**
-   * Returns (or issues connect and returns) transport channel by given transport {@code endpoint}.
-   * 
-   * @param endpoint by which to get or connect transport channel
-   * @throws IllegalArgumentException if {@code endpoint} is null
-   * @return non null transport object
+   * Stop transport, disconnect all available connections which belong to this transport. <br/>
+   * After transport is stopped it can't be opened again. Observable returned from method {@link #listen()} will
+   * immediately emit onComplete event for all subscribers.
    */
-  @Nonnull
-  ITransportChannel to(@CheckForNull TransportEndpoint endpoint);
+  void stop();
+
+  /**
+   * Stop transport, disconnect all available connections which belong to this transport. <br/>
+   * After transport is stopped it can't be opened again. Observable returned from method {@link #listen()} will
+   * immediately emit onComplete event for all subscribers. <br/>
+   * Stop is async operation, if result of operation is not needed leave second parameter null, otherwise pass
+   * {@link SettableFuture}.
+   *
+   * @param promise promise will be completed with result of closing (void or exception)
+   */
+  void stop(@Nullable SettableFuture<Void> promise);
+
+  /**
+   * Issues connect to the given transport address. This method may be used in case if specific incarnation id of remote
+   * endpoint is unknown to connect by address. So result endpoint then can be used for message sending.
+   *
+   * @param address address of transport endpoint to connect
+   * @return Listenable future to remote transport endpoint, which is completed once the handshake is passed.
+   */
+  ListenableFuture<TransportEndpoint> connect(@CheckForNull final TransportAddress address);
+
+  /**
+   * Disconnects transport channel to the given endpoint. If there is no connection do nothing. <br/>
```

Comment:
We should define what  will happen if we call disconnect(endpoint) and then send(endpoint, message) after.